.github/workflows/gen_pr.yaml

```diff
@@ -0,0 +1,54 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Run Generator and Create Pull Request
+on:
+  push:
+    branches:
+      - "master"
+jobs:
+  create_pr:
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Check out repo
+      uses: actions/checkout@v2
+    - name: Setup Node for gen/postgen scripts
+      uses: actions/setup-node@v2
+      with:
+        node-version: '12'
+    - name: Setup Go for proto2jsonschema script
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.14'
+    - name: Run the generators
+      run: ./scripts/gen.sh
+    - name: Commit changelog and manifest files
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        if [[ `git status --porcelain` ]]; then
+          # Changes after running generator
+          git config --global user.email "yoshi-code-bot@google.com"
+          git config --global user.name "Google Bot"
+          git checkout -b "pr_bot-$GITHUB_RUN_ID"
+          git add -A
+          NOW=$(date +"%m-%d-%Y")
+          git commit --message "feat: run generator (${NOW})"
+          git push origin "pr_bot-$GITHUB_RUN_ID"
+          echo "Created commit: $(git rev-parse HEAD)"
+          gh pr create --title "Run the generator (${NOW})" --body "AUTOMATED PR! This PR was created from a recent push to the default branch."
```

Comment:
nit: It would be good to add reviewer instructions to the body or a link to reviewer instructions in a README.

Reply:
I've added some more instructions in the description here.

We may further refine this as we test this in practice.